### DISCUSSION TITLE
RPH console compat.

### DISCUSSION THREAD

**Discussion by Michael21107** [scripthookvdotnet/scripthookvdotnet#1338] [Q&A] [answered]
Does RPH disable the console?

**Comment by kagikn** [comment]
Just don't use the same key for the both consoles, and you'll be able to use them without having difficulties. You'll need to set the `ConsoleKey` key to `Miscellaneous` section in RPH's ini to change the open/close key for RPH's console btw.

**Reply by Michael21107** [reply]
gotcha, thanks
I noticed that even by some F4, ESC, and Backspace combinations I can't accidentally open the SHVDN console with RPH when it's on the same key bind, is it just covered up or is it disabled cuz of the conflict?

**Reply by kagikn** [reply]
I'm not 100% sure what you mean, but you wouldn't want to use the same key for the SHVDN and RPH consoles. For example, if you use F4 to open or close RPH console, you wouldn't want to use a key combination with F4 (like `F4 + C`) to open or close SHVDN console. It would be appreciated if you show me examples that explains what you are saying.

**Reply by Michael21107** [reply]
i cant open both with the same key so they would overlap, i can only see RPH at that point so you wouldn't even realize the SHVDN console exists

**Reply by kagikn** [reply]
I know RPH and SHVDN use F4 as the default console key, and I'm guessing you asking me to change the default keybind. Is that what you are asking, correct?

**Reply by Michael21107** [reply]
I am asking if the SHVDN console is disabled if it somehow detects the RPH console opened or being set to the same key, or if it's just hidden behind it.

**Reply by kagikn** [reply, marked as answer]
Now I understand what you are asking better, and here's my answer; RPH blocks all scripts including external asi scripts (all `GtaThread`s strictly, I think) when its console is open, and therefore SHVDN can't do anything when RPH console is open. I see SHVDN console opens or closes at the same time when RPH console closes though. I think SHVDN misses a key press event sent by Windows when RPH console opens, but I don't consider it a bug of SHVDN or Windows.

**Reply by Michael21107** [reply]
gotcha, thank you
